std-bits/table/src/main/java/org/enso/table/data/column/operation/unary/SignumOperation.java

```diff
@@ -0,0 +1,114 @@
+package org.enso.table.data.column.operation.unary;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import org.enso.table.data.column.builder.Builder;
+import org.enso.table.data.column.operation.StorageIterators;
+import org.enso.table.data.column.operation.UnaryOperation;
+import org.enso.table.data.column.operation.map.MapOperationProblemAggregator;
+import org.enso.table.data.column.storage.ColumnDoubleStorage;
+import org.enso.table.data.column.storage.ColumnLongStorage;
+import org.enso.table.data.column.storage.ColumnStorage;
+import org.enso.table.data.column.storage.NullStorage;
+import org.enso.table.data.column.storage.type.BigDecimalType;
+import org.enso.table.data.column.storage.type.BigIntegerType;
+import org.enso.table.data.column.storage.type.FloatType;
+import org.enso.table.data.column.storage.type.IntegerType;
+import org.enso.table.data.column.storage.type.NullType;
+
+public class SignumOperation implements UnaryOperation {
+  public static final UnaryOperation INSTANCE = new SignumOperation();
+
+  @Override
+  public String getName() {
+    return "signum";
+  }
+
+  @Override
+  public boolean canApply(ColumnStorage<?> storage) {
+    return switch (storage.getType()) {
+      case IntegerType ignored -> true;
+      case FloatType ignored -> true;
+      case BigIntegerType ignored -> true;
+      case BigDecimalType ignored -> true;
+      case NullType ignored -> true;
+      default -> false;
+    };
+  }
+
+  @Override
+  public ColumnStorage<?> apply(
+      ColumnStorage<?> storage, MapOperationProblemAggregator problemAggregator) {
+    if (storage.getType() instanceof NullType) {
+      return new NullStorage(Math.toIntExact(storage.getSize()));
+    }
+
+    if (storage instanceof ColumnLongStorage columnLongStorage) {
+      return StorageIterators.mapOverLongStorage(
+          columnLongStorage,
+          Builder.getForLong(IntegerType.INT_8, storage.getSize(), problemAggregator),
+          (index, value, isNothing) -> signum(value));
+    }
+
+    if (storage instanceof ColumnDoubleStorage columnStorage) {
+      return StorageIterators.mapOverStorage(
+          columnStorage,
+          Builder.getForDouble(FloatType.FLOAT_64, storage.getSize(), problemAggregator),
```

Comment:
Why is the return type not `INT_8`? Shouldn't result of `signum` be either `-1` or `0` or `1`? E.g. always the smallest integers?

Reply:
Not a decision I could align with, but also not my business.